src/Client.c

```diff
@@ -294,8 +294,8 @@ MI_EXPORT MI_Uint32 WINAPI WSManCreateSession(
     {
         GOTO_ERROR("Failed to default http transport", miResult);
     }
-    /* Packet privacy over HTTP by default unless otherwise stated */
-    miResult = MI_DestinationOptions_SetPacketPrivacy(&(*session)->destinationOptions, MI_TRUE);
+    /* HTTP only supports unencrypted way now */
```

Comment:
It is set [WSMAN_OPTION_UNENCRYPTED_MESSAGES](https://github.com/PowerShell/psl-omi-provider/blob/2fbfb663f9a236c27eb298567c90386882c9e206/src/Client.c#L553) , and it is set in PowerShell [WSMAN_OPTION_UNENCRYPTED_MESSAGES](https://github.com/PowerShell/PowerShell/blob/c1c5344a8897262433141ecbc13bb06ac2c4bbef/src/System.Management.Automation/engine/remoting/fanin/WSManTransportManager.cs#L1532)

so we use encryption by default(MI_TRUE) for both http and https before, and if user special Noencryption option when remoting,  message will set to 1 with privacy MI_FALSE.

message set to 1, will set privacy to MI_FALSE: https://github.com/PowerShell/psl-omi-provider/blob/2fbfb663f9a236c27eb298567c90386882c9e206/src/Client.c#L558

now we should change to use noencryption by default(MI_FALSE) for both http and https, and also need to set message to 0 for -UseSSL on PowerShell side?

Another way is to set message to 1 when remoting with http on PowerShell side, then we can dismiss this PR, looks like this way will be better, how do you think?

Reply:
Now pwsh on Linux don't have New-PSSession command with NoEncryption, so user have no way to do it in this way now.

Reply:
My fix looks like makes everything use unencrypted way by default now.
If user use https, then if he don't set anything, it will use message 0 and with privacy MI_TRUE with encryption. if he set NoEncryption will make message to 1 and with privacy MI_FALSE with unencypted.

Reply:
let me test current fix for both http and https today to see how it goes, share the result with you later, thanks.